### PR TITLE
[kodi] Improvements in Kodi thing status handling

### PR DESCRIPTION
**Pull request by cweitkamp**
The thing status changed to OFFLINE if the Kodi instance is shut down, restarted or crashed. But it never was set to ONLINE again. This PR adds updates of the thing status whenever the Kodi instance sends a websocket message.

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java [hunk 1]
```diff
@@ -412,17 +412,18 @@ private void processApplicationStateChanged(String method, JsonObject json) {
...
 
     private void processSystemStateChanged(String method, JsonObject json) {
-        if ("System.OnQuit".equals(method) || "System.OnSleep".equals(method) || "System.OnRestart".equals(method)) {
-            try {
-                listener.updateConnectionState(false);
-            } catch (Exception e) {
-                logger.error("Event listener invoking error", e);
-            }
+        if ("System.OnQuit".equals(method) || "System.OnRestart".equals(method) || "System.OnSleep".equals(method)) {
+            listener.updateConnectionState(false);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java]: @cweitkamp is it somehow likely that the catch served a purpose, will things break now in certain cases. Good to remove it but did you consider which potential problems where hidden here?

Comment (cweitkamp) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java]: @martinvw I am glad you are asking. I thought about it and analysed the situation. Let me explain my doing:
1. There are a lot more lines of code which calls the same `updateConnectionState` method which aren't wrapped by a try catch block.
2. The definition of the method doesn't have a `throw`. So why catch something, which doesn't throw anything.
3. If you have a look at the implementation of the method you will see, that in the `false` part there is only a `updateStatus(ThingStatus.OFFLINE)`. This method needs no try catch block. And in the `true` part there is a `updateStatus(ThingStatus.ONLINE)` followed by another try catch block. Double usage.

Comment (martinvw) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/protocol/KodiConnection.java]: 1. okay, makes sense
2. Please note that a method can always throw a RuntimeException, whether it's annotated with a throws or not. 
3. that is the most convincing one :-)

**Comment by martinvw** [comment]
@pail23 any objections?

**Comment by pail23** [comment]
Lgtm.

**Pull request merged**